code.js

```diff
@@ -1,3 +1,20 @@
+// I THINK THIS IS CLOSE TO WHAT I NEED; HOWEVER, THE TESTING SUITE DOES NOT AGREE. I GOT A COUNTEREXAMPLE OF [] MULTIPLE TIMES WHEN ATTEMPTING TO RUN THIS
```

Comment:
Looks good -- the tests are failing because they are expecting the conversion function from matrix to list. This is fixed in the original repository, or you can change the tests accordingly.